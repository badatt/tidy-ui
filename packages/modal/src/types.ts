import { HTMLAttributes } from 'react';
import { IBaseProps, IClosable } from '@tidy-ui/types';

/**
 * The IModalProps interface equipping you with a robust set of properties to create dynamic and user-friendly modal components.
 * Whether you're displaying additional information, capturing user input, or providing context, the properties offered by
 * IModalProps will help you design modals that enhance user experience.
 *
 * @interface IModalProps
 * @type {IModalProps}
 * @augments {HTMLAttributes<HTMLDivElement>}
 * @public
 * @since 0.47.0
 */
export interface IModalProps extends IBaseProps, IClosable, HTMLAttributes<HTMLDivElement> {
  /**
   * Define a callback function that triggers when the backdrop of the modal is clicked. This allows you to implement custom behavior when the user interacts with the backdrop area.
   *
   * @type {Function}
   * @public
   * @since 0.47.0
   */
  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
  onBackdropClick?: (e?: any) => void;
}
