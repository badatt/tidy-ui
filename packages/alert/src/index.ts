export * from './Alert';
export * from './AlertFooter';
export * from './AlertTitle';
