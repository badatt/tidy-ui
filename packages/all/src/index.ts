export * from '@tidy-ui/commons';
export * from '@tidy-ui/feedback';
export * from '@tidy-ui/form';
export * from '@tidy-ui/layout';
export * from '@tidy-ui/navigation';
export * from '@tidy-ui/presentation';
