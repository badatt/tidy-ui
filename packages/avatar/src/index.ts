export * from './Avatar';
export * from './AvatarBadge';
export * from './AvatarGroup';
