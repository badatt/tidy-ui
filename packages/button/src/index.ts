export * from './Button';
export * from './Anchor';
