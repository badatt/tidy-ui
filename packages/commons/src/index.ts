export * from './actions';
export * from './hooks';
export * from './icons';
export * from './provider';
export * from './reducers';
export * from './theme';
export * from './utils';
