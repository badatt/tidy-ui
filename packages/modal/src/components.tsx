import { color, hsla, styled } from '@tidy-ui/commons';
import { IModalProps } from './types';

/**
 * Internal Modal root component
 *
 * @internal
 */
const ModalRoot = styled.div<IModalProps>`
  z-index: ${Number.MAX_SAFE_INTEGER};
  position: fixed;
  background-color: ${hsla(color.slate[900], 0.7)};
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
`;

/**
 * Internal ModalContent component
 *
 * @internal
 */
const ModalContent = styled.div<IModalProps>``;

export { ModalContent, ModalRoot };
