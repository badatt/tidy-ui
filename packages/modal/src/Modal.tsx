import React from 'react';
import { useIsMounted, usePortal } from '@tidy-ui/commons';
import { ModalContent, ModalRoot } from './components';
import { IModalProps } from './types';

/**
 * Modal can be used to show a component focused on the screen.
 * For example to show an alert, shades the background and highlights
 * only the component child under Modal
 */
const Modal = React.forwardRef<HTMLDivElement, IModalProps>((props, ref) => {
  const { children, isOpen, onBackdropClick, ...rest } = props;

  const isMounted = useIsMounted();
  const { Portal } = usePortal();

  const handleClose = React.useCallback(
    (e: React.MouseEvent<HTMLElement>) => {
      e.stopPropagation();
      if (isMounted()) {
        onBackdropClick?.(e);
      }
    },
    [isMounted, onBackdropClick],
  );

  if (!isOpen) {
    return null;
  }

  return (
    <Portal>
      <ModalRoot ref={ref} role="navigation" onClick={handleClose} {...rest}>
        <ModalContent onClick={(e) => e.stopPropagation()}>{children}</ModalContent>
      </ModalRoot>
    </Portal>
  );
});

Modal.defaultProps = {
  isOpen: false,
};

Modal.displayName = 'Modal';

export { Modal };
