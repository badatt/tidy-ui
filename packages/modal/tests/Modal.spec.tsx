/**
 * @jest-environment jsdom
 */
import React from 'react';
import { fireEvent, render, getByTestId } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { orchidDark, orchidLight, TidyUiProvider } from '../../commons/src';
import { Modal } from '../src';

describe('Modal', () => {
  it('Basic render', () => {
    const mockSetOpen = jest.fn();
    const mockSetClose = jest.fn();
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <button onClick={mockSetOpen} data-testid="open-btn">
          Submit
        </button>
        <Modal isOpen={true}>
          <div onClick={mockSetClose} data-testid="inner-element">
            Successfully submitted !
          </div>
        </Modal>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
    const openBtn = tree.getByTestId('open-btn');
    fireEvent.click(openBtn);
    expect(mockSetOpen).toHaveBeenCalledTimes(1);
    const innerElement = tree.getByTestId('inner-element');
    fireEvent.click(innerElement);
    expect(mockSetClose).toHaveBeenCalledTimes(1);
  });

  it('Dark mode basic render', () => {
    const tree = render(
      <TidyUiProvider theme={orchidDark}>
        <button data-testid="open-btn">Submit</button>
        <Modal isOpen={false}>
          <div>Successfully submitted !</div>
        </Modal>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('On backdrop click', () => {
    const mockSetOpen = jest.fn();
    const mockSetClose = jest.fn();
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <button onClick={mockSetOpen} data-testid="open-btn">
          Submit
        </button>
        <Modal isOpen onBackdropClick={mockSetClose}>
          <div>Successfully submitted !</div>
        </Modal>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
    const openBtn = tree.getByTestId('open-btn');
    fireEvent.click(openBtn);
    expect(mockSetOpen).toHaveBeenCalledTimes(1);
    const outerElement = tree.getByRole('navigation');
    fireEvent.click(outerElement);
    expect(mockSetClose).toHaveBeenCalledTimes(1);
  });

  it('On backdrop click empty', () => {
    const mockSetOpen = jest.fn();
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <button onClick={mockSetOpen} data-testid="open-btn">
          Submit
        </button>
        <Modal isOpen data-testid="outer-element">
          <div>Successfully submitted !</div>
        </Modal>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
    const openBtn = tree.getByTestId('open-btn');
    fireEvent.click(openBtn);
    expect(mockSetOpen).toHaveBeenCalledTimes(1);
    const outerElement = tree.getByRole('navigation');
    fireEvent.click(outerElement);
  });
});
