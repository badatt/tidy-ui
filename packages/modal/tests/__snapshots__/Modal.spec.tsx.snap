// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Basic render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  z-index: 9007199254740991;
  position: fixed;
  background-color: hsla(222,47%,11%,0.7);
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}

<body>
    <div>
      <button
        data-testid="open-btn"
      >
        Submit
      </button>
    </div>
    <div
      class="c0"
      role="modal"
    >
      <div
        class=""
      >
        <div
          data-testid="inner-element"
        >
          Successfully submitted !
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <button
      data-testid="open-btn"
    >
      Submit
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal Dark mode basic render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        data-testid="open-btn"
      >
        Submit
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      data-testid="open-btn"
    >
      Submit
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal On backdrop click 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  z-index: 9007199254740991;
  position: fixed;
  background-color: hsla(222,47%,11%,0.7);
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}

<body>
    <div>
      <button
        data-testid="open-btn"
      >
        Submit
      </button>
    </div>
    <div
      class="c0"
      data-testid="outer-element"
      role="modal"
    >
      <div
        class=""
      >
        <div>
          Successfully submitted !
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <button
      data-testid="open-btn"
    >
      Submit
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal On backdrop click empty 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  z-index: 9007199254740991;
  position: fixed;
  background-color: hsla(222,47%,11%,0.7);
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}

<body>
    <div>
      <button
        data-testid="open-btn"
      >
        Submit
      </button>
    </div>
    <div
      class="c0"
      data-testid="outer-element"
      role="modal"
    >
      <div
        class=""
      >
        <div>
          Successfully submitted !
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <button
      data-testid="open-btn"
    >
      Submit
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
